base/broadcast.jl

```diff
@@ -216,36 +245,36 @@ function broadcast_t(f, ::Type{Any}, As...)
         return similar(Array{Union{}}, shape)
     end
     nargs = length(As)
-    keeps, Idefaults = map_newindexer(shape, As)
+    keeps, Idefaults = map_indexer(shape, As)
     st = start(iter)
     I, st = next(iter, st)
-    val = f([ As[i][newindex(I, keeps[i], Idefaults[i])] for i=1:nargs ]...)
+    val = f(( _broadcast_getindex(As[i], newindex(I, keeps[i], Idefaults[i])) for i=1:nargs )...)
```

Comment:
No, this is applying the function being broadcasted to calculate de first element of the final array (and also uses its type as starting point for constructing the "final" array)